### PR TITLE
feat(DIA-902): login welcome step omnibus

### PR DESCRIPTION
**Pull request by iskounen**
This PR includes several tasks to polish the login welcome step, where the user enters their email address and is directed to either the sign-up or login flow.

- On forms with a single field (email, password) the continue button is disabled until the form is valid
- Error messages are no longer displayed as the user types
- The sign-up password form now includes instructions for a valid password
- Terms and Conditions and Privacy Policy links open a modal

https://github.com/user-attachments/assets/1a3b51be-44be-4039-a3af-95b10af41469

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Polishes email and password steps of auth2 - iskounen

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 15ab567ba8376fbf94a7c3169f049e69a240373e
</p>

File: src/app/Scenes/Onboarding/Auth2/scenes/LoginWelcomeStep.tsx [hunk 1]
```diff
@@ -70,7 +69,7 @@ export const LoginWelcomeStep: React.FC = () => {
             })
           }
 
-          resetForm()
+          resetForm({ values: { email } })
```
Comment (iskounen) [inline src/app/Scenes/Onboarding/Auth2/scenes/LoginWelcomeStep.tsx:72]: This allows users to see the email address they entered if they navigate back from the sign-up/login password step.

**Review by damassi — DISMISSED** [review]
Nice solution to this 👍

File: src/app/Scenes/Onboarding/Auth2/scenes/SignUpPasswordStep.tsx [hunk 2]
```diff
@@ -74,11 +84,29 @@ const SignUpPasswordStepForm: React.FC = () => {
...
+        onChangeText={(text) => {
+          handleChange("password")(text)
+        }}
+        onSubmitEditing={handleSubmit}
+      />
 
       <Spacer y={2} />
 
-      <Button block width={100} onPress={handleSubmit} disabled={!isValid} loading={isSubmitting}>
+      <Button block width={100} onPress={handleSubmit} loading={isSubmitting}>
```
Comment (iskounen) [inline src/app/Scenes/Onboarding/Auth2/scenes/SignUpPasswordStep.tsx:109]: We do not display the criteria for a valid password to users, so the only way they can know what those are is by submitting the form and seeing if there are any errors. That is why I removed the `disabled` property for this button.

Comment (iskounen) [reply src/app/Scenes/Onboarding/Auth2/scenes/SignUpPasswordStep.tsx:109]: Actually, I think ☝️ is silly. We should tell the user what's up.

Simulator Screenshot - iPhone 16 - 2024-10-10 at 06 52 40

File: src/app/Scenes/Onboarding/Auth2/scenes/SignUpPasswordStep.tsx [hunk 3]
```diff
@@ -74,11 +84,41 @@ const SignUpPasswordStepForm: React.FC = () => {
...
+        Password must be at least 8 characters and include a lowercase letter, uppercase letter, and
+        digit.
+      </Text>
 
       <Spacer y={2} />
 
-      <Button block width={100} onPress={handleSubmit} disabled={!isValid} loading={isSubmitting}>
+      <Button
+        block
+        width={100}
```
Comment (damassi) [inline src/app/Scenes/Onboarding/Auth2/scenes/SignUpPasswordStep.tsx:117]: Will probably want to remove this width

**Review by damassi — APPROVED** [review]

File: src/app/Scenes/Onboarding/Auth2/scenes/SignUpPasswordStep.tsx (hunk 2 above)
Comment (damassi) [reply src/app/Scenes/Onboarding/Auth2/scenes/SignUpPasswordStep.tsx]: Yes! Then validate on submit and keep the UI stable

File: src/app/Scenes/Onboarding/Auth2/scenes/SignUpPasswordStep.tsx (hunk 3 above)
Comment (iskounen) [reply src/app/Scenes/Onboarding/Auth2/scenes/SignUpPasswordStep.tsx:117]: This is what I get when I remove it. Should the width be handled in a different way?

Simulator Screenshot - iPhone 16 - 2024-10-10 at 07 37 02

Comment (damassi) [reply src/app/Scenes/Onboarding/Auth2/scenes/SignUpPasswordStep.tsx:117]: width = '100%'! Just not a fixed pixel value; lean on dynamic screen dimensions.

**Pull request merged**